### DISCUSSION TITLE
URL link previews for site use incorrect title/metadata

### DISCUSSION THREAD

**Discussion by chasemccoy** [mastodon/mastodon#31287] [Q&A] [answered]
I automatically post links to new blog posts from my website (https://chasem.co) to my Mastodon profile (https://mastodon.social/@chsmc). The latest post renders a link preview correctly:

<img width="605" alt="image" src="https://github.com/user-attachments/assets/4e27a949-2792-49b0-b6ce-3200b71f6d76">

But the weird thing is that, when posting a new link, all of the past link previews for _different_ blog posts take on the title of the latest one:

<img width="622" alt="image" src="https://github.com/user-attachments/assets/e8c9e89a-e249-49e8-987a-176a81c18ba8">

All of those links go to separate pages with separate titles, and I've tested that their OG tags in the <head> are correct. This issue is seen in third party apps as well. 

I don't believe this is an issue with the metadata on my site, so I have to assume it's a Mastodon bug? But would appreciate any ideas for how to resolve it.

**Comment by c960657** [comment, marked as answer]
The page https://chsmc.org/2024/08/prowl-and-defiance/ contains the following:
```
<meta property="og:url" content="https://chsmc.org/">
```
i.ei. it states that the canonical URL of that page is https://chasem.co, and hence the two pages share the same preview card.

You should update the `og:url` attribute to reflect the URL of the current page.

**Reply by chasemccoy** [reply]
@c960657 Ah, nice catch! I totally misunderstood how that tag was supposed to work. Thanks for taking a look and letting me know!
